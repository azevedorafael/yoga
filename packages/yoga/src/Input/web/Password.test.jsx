import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { ThemeProvider, Input } from '../..';

describe('<Input.Password />', () => {
  describe('Snapshots', () => {
    it('should match with default input', () => {
      const { container } = render(
        <ThemeProvider>
          <Input.Password />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with disabled input', () => {
      const { container } = render(
        <ThemeProvider>
          <Input.Password disabled />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match with full width', () => {
      const { container } = render(
        <ThemeProvider>
          <Input.Password label="Label" full />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should call onChange', () => {
      const onChangeMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input.Password
            label="Input"
            data-testid="input"
            onChange={onChangeMock}
          />
        </ThemeProvider>,
      );

      fireEvent.change(getByTestId('input'), { target: { value: 'foo' } });

      expect(onChangeMock).toHaveBeenCalled();
    });

    it('should call onFocus', () => {
      const onFocusMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Input.Password
            label="Input"
            data-testid="input"
            onFocus={onFocusMock}
          />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));

      expect(onFocusMock).toHaveBeenCalled();
    });

    it('should call onBlur', () => {
      const onBlurMock = jest.fn();

      const { getByTestId } = render(
        <ThemeProvider>
          <Input.Password
            label="Input"
            data-testid="input"
            onBlur={onBlurMock}
          />
        </ThemeProvider>,
      );

      fireEvent.focus(getByTestId('input'));
      fireEvent.blur(getByTestId('input'));

      expect(onBlurMock).toHaveBeenCalled();
    });
  });

  describe('visibility button', () => {
    it('should toggle password visibility when click visibility button', () => {
      const { getByRole, getByTestId } = render(
        <ThemeProvider>
          <Input.Password label="Input" data-testid="input" />
        </ThemeProvider>,
      );

      expect(getByTestId('input').type).toBe('password');

      fireEvent.click(getByRole('button'));

      expect(getByTestId('input').type).toBe('text');
    });
  });
});
