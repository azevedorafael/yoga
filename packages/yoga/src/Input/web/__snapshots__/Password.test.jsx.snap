// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input.Password /> Snapshots should match with default input 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c2 {
  width: 320px;
}

.c2 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A6;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c2 svg:hover,
.c2 svg:focus {
  fill: #41414A;
}

.c2:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c2:hover fieldset,
.c2:focus-within fieldset {
  border-color: #41414A;
}

.c4 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c4:focus {
  color: #41414A;
}

.c4:focus + label {
  top: 0px;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c4:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c4::-webkit-input-placeholder {
  color: #9898A6;
}

.c4::-moz-placeholder {
  color: #9898A6;
}

.c4:-ms-input-placeholder {
  color: #9898A6;
}

.c4::placeholder {
  color: #9898A6;
}

.c4[type="number"]::-webkit-outer-spin-button,
.c4[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c4:invalid {
  box-shadow: none;
}

.c6 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c7 {
  visibility: hidden;
  position: relative;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  padding: 0;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  padding-left: 4px;
  padding-right: 4px;
}

.c5 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A6;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c1 {
  width: auto;
}

.c3 input {
  padding-right: 40px;
}

.c0 {
  position: relative;
  display: inline-block;
}

.c9 {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  cursor: pointer;
  outline: none;
}

.c9:hover svg,
.c9:focus svg {
  fill: #41414A;
}

.c9 svg {
  height: 52px;
  width: 20px;
  fill: #9898A6;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        label="Label"
      >
        <input
          class="c4"
          type="password"
          value=""
        />
        <label
          class="c5"
        >
          Label
        </label>
        <fieldset
          class="c6"
        >
          <legend
            class="c7"
          >
            <span
              class="c8"
            >
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="c9"
      role="button"
      tabindex="0"
    >
      <svg />
    </div>
  </div>
</div>
`;

exports[`<Input.Password /> Snapshots should match with disabled input 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
  border-color: #D3D3E2;
  color: #D3D3E2;
}

.c2 {
  width: 320px;
}

.c2 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A6;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c2 svg:hover,
.c2 svg:focus {
  fill: #41414A;
}

.c2:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c2:hover fieldset,
.c2:focus-within fieldset {
  border-color: #41414A;
  border-color: #D3D3E2;
}

.c2 svg {
  fill: #D3D3E2;
  pointer-events: none;
}

.c2 fieldset {
  border-color: #D3D3E2;
}

.c4 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c4:focus {
  color: #41414A;
}

.c4:focus + label {
  top: 0px;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c4:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c4::-webkit-input-placeholder {
  color: #9898A6;
}

.c4::-moz-placeholder {
  color: #9898A6;
}

.c4:-ms-input-placeholder {
  color: #9898A6;
}

.c4::placeholder {
  color: #9898A6;
}

.c4[type="number"]::-webkit-outer-spin-button,
.c4[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c4:invalid {
  box-shadow: none;
}

.c6 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c7 {
  visibility: hidden;
  position: relative;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  padding: 0;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  padding-left: 4px;
  padding-right: 4px;
}

.c5 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A6;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  color: #D3D3E2;
}

.c1 {
  width: auto;
}

.c3 input {
  padding-right: 40px;
}

.c0 {
  position: relative;
  display: inline-block;
}

.c9 {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  cursor: pointer;
  outline: none;
  cursor: not-allowed;
  pointer-events: none;
}

.c9:hover svg,
.c9:focus svg {
  fill: #41414A;
}

.c9 svg {
  height: 52px;
  width: 20px;
  fill: #9898A6;
  fill: #D3D3E2;
  pointer-events: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        disabled=""
        label="Label"
      >
        <input
          class="c4"
          disabled=""
          type="password"
          value=""
        />
        <label
          class="c5"
          disabled=""
        >
          Label
        </label>
        <fieldset
          class="c6"
          disabled=""
        >
          <legend
            class="c7"
          >
            <span
              class="c8"
            >
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="c9"
      disabled=""
      role="button"
    >
      <svg />
    </div>
  </div>
</div>
`;

exports[`<Input.Password /> Snapshots should match with full width 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
}

.c2 {
  width: 100%;
}

.c2 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A6;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c2 svg:hover,
.c2 svg:focus {
  fill: #41414A;
}

.c2:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c2:hover fieldset,
.c2:focus-within fieldset {
  border-color: #41414A;
}

.c4 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
}

.c4:focus {
  color: #41414A;
}

.c4:focus + label {
  top: 0px;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c4:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c4::-webkit-input-placeholder {
  color: #9898A6;
}

.c4::-moz-placeholder {
  color: #9898A6;
}

.c4:-ms-input-placeholder {
  color: #9898A6;
}

.c4::placeholder {
  color: #9898A6;
}

.c4[type="number"]::-webkit-outer-spin-button,
.c4[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c4:invalid {
  box-shadow: none;
}

.c6 {
  position: absolute;
  top: -7px;
  left: 0;
  padding: 0;
  padding-left: 0px;
  right: 0;
  bottom: 0;
  margin: 0;
  pointer-events: none;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
  padding-left: 12px;
}

.c7 {
  visibility: hidden;
  position: relative;
  max-width: 0.01px;
  width: auto;
  height: 14px;
  padding: 0;
  font-weight: normal;
  -webkit-transition: max-width ease 50ms;
  transition: max-width ease 50ms;
  max-width: 0.01px;
  font-size: 12px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  padding-left: 4px;
  padding-right: 4px;
}

.c5 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  font-size: 14px;
  font-weight: 400;
  color: #9898A6;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

.c1 {
  width: 100%;
}

.c3 input {
  padding-right: 40px;
}

.c0 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c9 {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  cursor: pointer;
  outline: none;
}

.c9:hover svg,
.c9:focus svg {
  fill: #41414A;
}

.c9 svg {
  height: 52px;
  width: 20px;
  fill: #9898A6;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
        label="Label"
      >
        <input
          class="c4"
          type="password"
          value=""
        />
        <label
          class="c5"
        >
          Label
        </label>
        <fieldset
          class="c6"
        >
          <legend
            class="c7"
          >
            <span
              class="c8"
            >
              Label
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="c9"
      role="button"
      tabindex="0"
    >
      <svg />
    </div>
  </div>
</div>
`;
